### PR TITLE
Deduplicate checkbox type components code

### PR DESCRIPTION
**Pull request by jtomasek**
~~Depends on https://github.com/openshift/console/pull/3516~~

### CODE REVIEW

File: frontend/packages/console-shared/src/components/formik-fields/SwitchField.tsx [hunk 1]
```diff
@@ -1,40 +1,15 @@
...
-        aria-describedby={`${fieldId}-helper`}
-        onChange={(value, event) => field.onChange(event)}
-      />
-    </FormGroup>
-  );
-};
+const SwitchField: React.FC<CheckboxFieldProps> = (baseProps) => (
+  <ToggleableFieldBase {...baseProps}>
+    {(props) => {
+      delete props.isValid;
```
Comment (christianvogt) [inline frontend/packages/console-shared/src/components/formik-fields/SwitchField.tsx]: Why is this here?

If absolutely necessary it's better to not modify props.

Do this instead.
`({ isValid, ...props }) => {`

Comment (jtomasek) [reply frontend/packages/console-shared/src/components/formik-fields/SwitchField.tsx]: `isValid` is not used on Switch PF4 component and it passes rest of the props to a DOM element, so it results in `backend.js:6 Warning: React does not recognise the `isValid` prop on a DOM element.` error. 

Fixed.
